python/morpheus/morpheus/pipeline/stage_decorator.py

```diff
@@ -72,6 +73,35 @@ def _validate_keyword_arguments(fn_name: str,
                              f"{fn_name} contains '{param.name}' that was not provided with a value")
 
 
+def _fn_receives_subscription(signature: inspect.Signature) -> bool:
+    """
+    Inspects a function's signature to determine if the function expects a `mrc.Subscription` as it's first argument.
+    This requires the type-hint to be set to either `mrc.Subscription` or a string hint of `"mrc.Subscription"` or
+    `"Subscription"`
+
+    Parameters
+    ----------
+    signature : `inspect.Signature` | `typing.Callable`
+        Function or signature of a function to be checked
+
+    Returns
+    -------
+    bool
+        True if the signature indicates the first argument to be a `mrc.Subscription`.
+    """
+    param_iter = iter(signature.parameters.values())
+
+    try:
+        first_param = next(param_iter)
+        if (first_param.annotation is mrc.Subscription or first_param.annotation == "mrc.Subscription"
+                or first_param.annotation == "Subscription"):
+            return True
```

Comment:
Why did you do it this way instead of using a for loop?
```python
for first_param in signature.parameters.values():
    ...
```

Reply:
I really am only interested in the first parameter, not all of the parameters.